test/e2e/helpers.js

```diff
@@ -564,37 +568,85 @@ const onboardingPinExtension = async (driver) => {
   await driver.clickElement('[data-testid="pin-extension-done"]');
 };
 
-const onboardingCompleteWalletCreationWithOptOut = async (driver) => {
+/**
+ * Completes the onboarding flow with optional opt-out settings for wallet creation.
+ *
+ * This function navigates through the onboarding process, allowing for opt-out of certain features.
+ * It waits for the appropriate heading to appear, then proceeds to opt-out of third-party API
+ * integration for general and assets sections if specified in the optOutOptions.
+ *
+ * @param {WebDriver} driver - The Selenium WebDriver instance.
+ * @param {object} optOutOptions - Optional. An object specifying which features to opt-out of.
+ * @param {boolean} optOutOptions.basicFunctionality - Optional. Defaults to true. Opt-out of basic functionality.
+ * @param {boolean} optOutOptions.profileSync - Optional. Defaults to true. Opt-out of profile sync.
+ * @param {boolean} optOutOptions.assets - Optional. Defaults to true. Opt-out of assets options.
+ * @param {boolean} optOutOptions.isNewWallet - Optional. Defaults to true. Indicates if this is a new wallet creation.
+ */
+const onboardingCompleteWalletCreationWithOptOut = async (
+  driver,
+  optOutOptions = {},
+) => {
+  const defaultOptOutOptions = {
+    basicFunctionality: true,
+    profileSync: true,
+    assets: true,
+    isNewWallet: true,
+  };
+
+  const optOutOptionsToUse = { ...defaultOptOutOptions, ...optOutOptions };
+
   // wait for h2 to appear
-  await driver.findElement({ text: 'Congratulations!', tag: 'h2' });
+  await driver.findElement({
+    text: optOutOptionsToUse.isNewWallet
+      ? 'Congratulations'
+      : 'Your wallet is ready',
+    tag: 'h2',
```

Comment:
is there a way to make this more non-tag dependent? If product decides that instead of a h1 tag is now something else, this test would be broken. Could we do that?

Reply:
I can make this change in a different PR. Don't want to make too many possible breaking changes as this is a requirement for a release